### ISSUE TITLE
Headers in link is not considered in resolve fully

### ISSUE THREAD

**Issue by ymohdriz** [swagger-api/swagger-parser#936]
Hi team,

Links is considered in resolve fully option and headers field in link is not considered in resolve fully. 

[issue_884.txt](https://github.com/swagger-api/swagger-parser/files/2588315/issue_884.txt)

Thanks,
Mohammed

**Comment by NathanAtKadaster** [comment]
So I am having similar issues where a valid spec does not contain any headers whatsoever, but what you attached here is not a valid spec. 

Please paste this in an editor like [editor.swagger.io](swagger editor) and you will see that the `UserRepository` schema is not correct:
```
    Schema error at components.links.userRepository
    should NOT have additional properties
    additionalProperty: operationId, headers
```

Can you specify why this spec should be working and how it is used? Maybe we can help eachother out.

**Comment by gracekarina** [comment]
Fixed by #937
